File: back/src/main/java/com/back/domain/user/dto/TicketGroupInfo.java
package com.back.domain.user.dto;

import com.back.domain.ticket.entity.Ticket;

import java.util.List;

public record TicketGroupInfo(
        Long scheduleId,
        String concertName,
        String urlPoster,
        String startDate,
        String endDate,
        int round,
        int totalPrice,
        List<TicketSummary> tickets
) {
    public static TicketGroupInfo from(List<Ticket> tickets) {
        Ticket first = tickets.get(0);
        return new TicketGroupInfo(
                first.getSchedule().getScheduleId(),
                first.getSchedule().getConcert().getConcertName(),
                first.getSchedule().getConcert().getUrlPoster(),
                first.getSchedule().getConcert().getStartDate().toLocalDate().toString(),
                first.getSchedule().getConcert().getEndDate().toLocalDate().toString(),
                first.getSchedule().getRound(),
                tickets.stream().mapToInt(Ticket::getTicketPrice).sum(),
                tickets.stream().map(TicketSummary::from).toList()
        );
    }
}

File: back/src/main/kotlin/com/back/domain/user/dto/TicketGroupInfo.kt
package com.back.domain.user.dto

import com.back.domain.ticket.entity.Ticket

data class TicketGroupInfo(
    val scheduleId: Long,
    val concertName: String,
    val urlPoster: String?,
    val startDate: String,
    val endDate: String,
    val round: Int,
    val totalPrice: Int,
    val tickets: List<TicketSummary>
) {
    companion object {
        fun from(tickets: List<Ticket>): TicketGroupInfo {
            require(tickets.isNotEmpty()) { "Tickets list must not be empty" }
            val first = tickets.first()
            val schedule = first.schedule
            val concert = schedule.concert

            return TicketGroupInfo(
                scheduleId = checkNotNull(schedule.scheduleId) { "Schedule ID null" },
                concertName = concert.concertName,
                urlPoster = concert.urlPoster,
                startDate = concert.startDate.toLocalDate().toString(),
                endDate = concert.endDate.toLocalDate().toString(),
                round = schedule.round,
                totalPrice = tickets.sumOf { it.ticketPrice },
                tickets = tickets.map { TicketSummary.from(it) }
            )
        }
    }
}
